### DISCUSSION TITLE
Labeled over 2000 frames and still getting low OKS

### DISCUSSION THREAD

**Discussion by aam112** [talmolab/sleap#1423] [Help!]
Hi,

I am training a single mouse model and have labeled over 2000 frames. However, the model evaluation is still suggesting that the pose estimation is not accurate. 

Do you have any suggestions for what I could do to improve the model?

<img width="1406" alt="Screenshot 2023-07-28 at 9 52 16 AM" src="https://github.com/talmolab/sleap/assets/97302592/ea92e796-226e-463c-95f0-05b7b59f9451">

<img width="688" alt="Picture1" src="https://github.com/talmolab/sleap/assets/97302592/9f4f5ab7-819b-4ee8-830a-bff530204dd7">

**Comment by roomrys** [comment]
Hi @aam112,

With 2000 frames labeled, we should be looking at your model parameters to make improvements. Would you mind sharing a screenshot of the training pipeline GUI (showing the model parameters)? What I can tell from the image alone is that it is quite pixelated, thus, I would refrain from doing much input scaling (0.5 to 0.8 minimum), especially if the mouse moves towards the back of the cage.

Qualitatively, are the predictions poor?

Thanks,
Liezl

**Reply by aam112** [reply]
Hi Liezl,

Here are the model parameters:
<img width="1357" alt="Screenshot 2023-08-02 at 10 09 27 AM" src="https://github.com/talmolab/sleap/assets/97302592/a8c8ac30-e66c-44b2-9637-e10974f4b11d">
<img width="1353" alt="Screenshot 2023-08-02 at 10 09 19 AM" src="https://github.com/talmolab/sleap/assets/97302592/75f824d9-c438-439a-a130-6bc9fd2c13dd">

The qualitative predictions are mostly okay but the tail is sometimes not labeled or is completely off and the legs don't seem to be moving properly in a lot of segments.

**Reply by roomrys** [reply]
Hi @aam112,

Hmm ok, for the single animal pipeline it is pretty important that the receptive field (the blue box overlaid on the image in the training gui) is about the size of the animal. If the receptive field is too small then that would result in trouble locating extremities and might explain why we see the tail disappearing (other than tails just being difficult to locate). 

We could try increasing the receptive field size by increasing the max stride. The input scaling also affects receptive field size, but I would leave that at 1.00 based on the resolution. 

Let me know if that helps! Otherwise, a few example screenshots of common errors might also be helpful.

Thanks,
Liezl

**Reply by aam112** [reply]
Hi, I reran the model and the metrics don't seem much different. 
<img width="1223" alt="Screenshot 2023-08-08 at 9 01 16 AM" src="https://github.com/talmolab/sleap/assets/97302592/b51bbefd-7dda-4bda-83b4-9f4d228988a1">
<img width="434" alt="Screenshot 2023-08-08 at 8 59 37 AM" src="https://github.com/talmolab/sleap/assets/97302592/17f35921-4326-44e0-962a-b870d43a7159">
<img width="641" alt="Screenshot 2023-08-08 at 9 00 45 AM" src="https://github.com/talmolab/sleap/assets/97302592/34e72c3e-090d-4dd2-9df7-c4d49db2c51c">

Here are some examples where the tail isn't tracked very well. Is this just because of the resolution of the videos? A mAP score as low as mine suggests that the model I trained is not tracking very accurately, right? I don't understand the evaluation metrics very well but from what I read the reported values in literature are closer to 0.7 or 0.8.

Thank you for your help!

**Reply by roomrys** [reply]
Hi @aam112,

Despite the OKS value being lower than you would like, my personal recommendation is to continue onto the analysis step as these predictions are quite good. The OKS value will be greatly influenced by the outliers in the error prone tail tip, hence the lower OKS.

Sorry I couldn't be of more help....

Thanks,
Liezl
